cmd/spire-server/cli/agent/list.go

```diff
@@ -81,57 +81,55 @@ func (c *listCommand) Run(ctx context.Context, env *common_cli.Env, serverClient
 		if err != nil {
 			return err
 		}
-		agents = append(agents, listResponse.Agents...)
+		response.Agents = append(response.Agents, listResponse.Agents...)
 		if pageToken = listResponse.NextPageToken; pageToken == "" {
 			break
 		}
 	}
 
-	if len(agents) == 0 {
-		return env.Printf("No attested agents found\n")
-	}
-
-	msg := fmt.Sprintf("Found %d attested ", len(agents))
-	msg = util.Pluralizer(msg, "agent", "agents", len(agents))
-	env.Printf(msg + ":\n\n")
+	c.printer.MustPrintProto(response)
 
-	return printAgents(env, agents...)
+	return nil
 }
 
 func (c *listCommand) AppendFlags(fs *flag.FlagSet) {
 	fs.StringVar(&c.matchSelectorsOn, "matchSelectorsOn", "superset", "The match mode used when filtering by selectors. Options: exact, any, superset and subset")
 	fs.Var(&c.selectors, "selector", "A colon-delimited type:value selector. Can be used more than once")
+	cliprinter.AppendFlagWithCustomPretty(&c.printer, fs, c.env, prettyPrintAgents)
+}
+
+func prettyPrintAgents(env *commoncli.Env, results ...interface{}) error {
+	agents := results[0].(*agentv1.ListAgentsResponse).Agents
+
+	msg := fmt.Sprintf("Found %d attested ", len(agents))
+	msg = util.Pluralizer(msg, "agent", "agents", len(agents))
+	env.Printf("%s:\n\n", msg)
+	return printAgents(env, agents...)
 }
 
-func printAgents(env *common_cli.Env, agents ...*types.Agent) error {
+func printAgents(env *commoncli.Env, agents ...*types.Agent) error {
+	if len(agents) == 0 {
```

Comment:
change in behavior? This used to happen before we printed "Found X attested agent(s)", but now it happens after.